federated_aws_cli/login.py

```diff
@@ -200,33 +221,24 @@ def callback(self, code, state, token=None, **kwargs):
         logger.debug(
             'Roles and aliases are {}'.format(self.role_map))
 
-        credentials = message = None
-
         # TODO: Consider whether this needs to loop forever
-        while credentials is None:
+        while self.credentials is None:
             # If we don't have a role ARN on the command line, we need to show
             # the role picker
             if self.role_arn is None and not self.batch:
-                try:
-                    self.role_arn = show_role_picker(self.role_map, message)
-                except NoPermittedRoles as e:
-                    logger.error(e)
-                    exit_sigint()
-                logger.debug('Role ARN {} selected'.format(self.role_arn))
-
-            # If they somehow exit out of the role picker or there aren't
-            # any choices
-            if self.role_arn is None:
-                logger.info('Exiting, no IAM Role ARN selected')
-                exit_sigint()
+                self.state = "role_picker"
+
+                while not self.role_arn:
+                    time.sleep(.05)
```

Comment:
Out of curiosity, what is this for? Might be worth adding a comment about what's going on here.